src/util-decode-mime.c

```diff
@@ -1265,21 +1265,39 @@ static uint32_t ProcessBase64Remainder(
     return buf_consumed;
 }
 
+/**
+ * \brief Skip any characters outside of the base64 alphabet as RFC 2045
+ * is followed for MIME parsing. Also, update bvr_len accordingly as if the
+ * DecodeBase64 returned more than 4 Bytes in there, it is likely because
+ * of the presence of invalid characters in the encoded string.
+ *
+ * \param buf The current line
+ * \param len The length of the line
+ * \param offset The offset where to start reading the data from
+ * \param state The current parser state
+ * */
+static inline void B64SkipUnneededCharsLastBlk(
+        const uint8_t *buf, const uint32_t buf_len, const uint32_t offset, MimeDecParseState *state)
+{
+    for (uint32_t i = offset; i < buf_len; i++) {
+        if (IsBase64Alphabet(buf[i])) {
+            state->bvremain[state->bvr_len++] = buf[i];
```

Comment:
I would sleep better if this loop would have a guard against overflowing `bvremain`.